### PR TITLE
Feature - Migrations use mocked data factories

### PR DESCRIPTION
**Pull request by priesgo**
* All migrations are now using the mocked data factories
* Many issues on migration scripts arised when using this mocked data. Some of them due to nullable fields being null in the current mocked data and some others due to not having empty strings on non nullable fields.

Also several glitches migrating between different versions were identified I will report those separately.

@Antonior26 
@greglever

### CODE REVIEW

**Comment by priesgo** [comment]
Almost forget... Cancer exit questionnaire was also missing from the latest model version...

File: protocols/migration/base_migration.py [hunk 1]
```diff
@@ -10,3 +10,12 @@ def validate_object(object_to_validate, object_type):
...
             raise Exception("New {object_type} object is not valid".format(object_type=object_type))
+
+    @staticmethod
+    def convert_string_to_integer(string):
+        if string is None:
+            return None
+        try:
+            return int(string)
+        except ValueError:
+            raise Exception("Value: {string} is not an integer contained in a string !".format(string=string))
```
Comment (greglever) [inline protocols/migration/base_migration.py]: Can we please put this in `protocols/util`. Maybe in something like `string_util.py`

File: protocols/tests/test_migration/test_migration_reports_3_0_0_to_participant_1_0_0.py [hunk 2]
```diff
@@ -1,22 +1,24 @@
 from unittest import TestCase
 
-from protocols.util import get_valid_cancer_participant_3_0_0
-from protocols.util import get_valid_cancer_participant_1_0_0
 from protocols.reports_3_0_0 import CancerParticipant as CancerParticipant_old
 from protocols.participant_1_0_0 import CancerParticipant as CancerParticipant_new
 from protocols.migration.migration_reports_3_0_0_to_participant_1_0_0 import MigrateReports3ToParticipant1
+from protocols.util.factories.avro_factory import GenericFactoryAvro
+from protocols.util.dependency_manager import VERSION_400, VERSION_300
```
Comment (greglever) [inline protocols/tests/test_migration/test_migration_reports_3_0_0_to_participant_1_0_0.py]: Can we please change this line to:
```
from protocols.util.dependency_manager import VERSION_400
from protocols.util.dependency_manager import VERSION_300
```
or 
```
from protocols.util import dependency_manager
```
and refer to `dependency_manager.VERSION_400` and `dependency_manager.VERSION_300`

File: protocols/tests/test_migration/test_migration_reports_3_0_0_to_reports_4_0_0.py [hunk 3]
```diff
@@ -5,22 +5,21 @@
 from protocols.reports_3_0_0 import ReportedSomaticVariants as ReportedSomaticVariants_old
 from protocols.reports_4_0_0 import ReportEventCancer as ReportEventCancer_new
 from protocols.reports_4_0_0 import ReportedSomaticVariants as ReportedSomaticVariants_new
-from protocols.util import get_valid_reported_somatic_variant_3_0_0
-from protocols.util import get_valid_reported_somatic_variant_4_0_0
-from protocols.util.generate_mock_objects import get_valid_report_event_cancer_3_0_0
+from protocols.util.dependency_manager import VERSION_300, VERSION_400
```
Comment (greglever) [inline protocols/tests/test_migration/test_migration_reports_3_0_0_to_reports_4_0_0.py]: Can we please change this line to:
```
from protocols.util.dependency_manager import VERSION_400
from protocols.util.dependency_manager import VERSION_300
```
or 
```
from protocols.util import dependency_manager
```
and refer to `dependency_manager.VERSION_400` and `dependency_manager.VERSION_300`

File: protocols/tests/test_migration/test_migration_reports_4_0_0_to_reports_4_2_0.py [hunk 4]
```diff
@@ -4,6 +4,8 @@
 from protocols import reports_4_2_0
 from protocols.migration import MigrateReports400To420
 from protocols.util import get_valid_cancer_interpretation_request_4_0_0
+from protocols.util.dependency_manager import VERSION_310, VERSION_400
```
Comment (greglever) [inline protocols/tests/test_migration/test_migration_reports_4_0_0_to_reports_4_2_0.py]: Can we please change this line to:
```
from protocols.util.dependency_manager import VERSION_400
from protocols.util.dependency_manager import VERSION_300
```
or 
```
from protocols.util import dependency_manager
```
and refer to `dependency_manager.VERSION_400` and `dependency_manager.VERSION_300`

File: protocols/tests/test_migration/test_migration_reports_4_0_0_to_reports_4_2_0.py [hunk 5]
```diff
@@ -13,7 +15,10 @@ class TestMigrateReports420To4(TestCase):
 
     def test_migrate_cir_400_to_420(self):
 
-        cir_400 = get_valid_cancer_interpretation_request_4_0_0()
+        #cir_400 = get_valid_cancer_interpretation_request_4_0_0()
```
Comment (greglever) [inline protocols/tests/test_migration/test_migration_reports_4_0_0_to_reports_4_2_0.py]: can we remove this comment line please

**Pull request merged**